#### 


#### steps

* mapping Hi-C reads from each sample to co-assembly

* Plasmid contigs:
    1. blast plasmid against V583, HIP genome, replicon and PLS-DB (plasmid database)
    
        1. filter criteria: identity > 90, length > 800 bp, matched length/ contig length > 0.6 
    2. candidate contig results: blast against nr data base using webpage blast and manually checked the results, removed those having hits on other genomes
    3. Results: 29 contigs with 6 carrying Rep
* Res contig:
    1. using ResFinder

* check HiC links in each sample by plasmid or ARG       
    1. find some contigs connected with EF but not clustered in dRep results, gp back to bin3C results, many of them are in EF cluster by bin3C results
    2. add thost contigs back into EF bin (linked with ef plasmid > 4 and binned into EF by bin3c)
    3. ~ 80 bins in EF in dRep and 208 in EF in bin3c, check completedness and contamination rate before and after adding contigs from bin3C

| bin    | size | completedness | contamination rate | Strain heterogeneity |
|--------|------|---------------|--------------------|----------------------|
| before | 82   | 98.41         | 0.37               | 50                   |
| after  | 141  | 99.44         | 0.94               | 50.00                |

* HiC links summarized


|                | V583    |         | HIP     |         | VE      |         |
|----------------|---------|---------|---------|---------|---------|---------|
| Day            | 0       | 13      | 0       | 13      | 0       | 13      |
| total links    | 2998446 | 5364364 | 2655326 | 9029966 | 2612002 | 4363302 |
| non-self links | 2789944 | 4513821 | 2447986 | 8371705 | 2430636 | 3690062 |
| plasmid links  | 4       | 174     | 0       | 1178    | 2       | 1688    |
| non clustered  | 2       | 32      | 0       | 362     | 2       | 318     |
| ARG links      | 484     | 936     | 738     | 5352    | 468     | 986     |
| non clustered  | 302     | 356     | 374     | 2990    | 232     | 374     |


* draw heatmap showing HiC links between bins and plasmid/ARG contigs
    1. Hi-C links were normalized by the (absolute) abundance of the bin (genome) Transcripts Per Million ? or NumReads?
    2.  how about length?? should it be length * abundance = reads??


#### Things to do

1. check Vancomycin location of pTEF plasmid and the changes caused by plasmid introduction.

    - pTEF1 and pTEF2 are structurally similar to the archetypal pheromone responsive plasmids pAD1 (15) and pCF10 (14), respectively, and pTEF3 belongs to the family of pAMÎ²1 broad host range plasmids.
    - The unique regions in pTEF1 include a Tn4001-like transposon encoding aminoglycoside resistance and another IS-flanked element carrying erythromycin resistance and multidrug resistance genes.
    - V583: VanB-Tn1549

2. Complare the plasmid linked E.f contigs in 3 samples (it is wired that VE has more plasmid links)

    - more plasmid contigs in VE???
    - plasmid contig abundance in three samples (TPM, normalized by total reads?)
    - Hi-C links of contigs - bin in three samples
    - Bin abundance


3. Check the insertion genes (flanking genes) of resistance genes, if the sequence changed over time?? usingi individual assembly


ClobberError: This transaction has incompatible packages due to a shared path.
  packages: defaults::matplotlib-2.2.3-py27hb69df0a_0, conda-forge::matplotlib-base-2.2.4-py27hfd891ef_0
  path: 'lib/python2.7/site-packages/pylab.pyc'

4. prepare replicon and resistance gene region on contigs in bed format (coassembly)



`samtools view -b -L region.bed test.bam > test_region.bam`

sort and index each bam file, and then extract rep region

####Day 13, HIP

| contig       | Links_number | ref           | replicon |
|--------------|--------------|---------------|----------|
| k141_813771  | 1            | pTEF1+VE14089 | Rep9     |
| k141_2242115 | 13           | pTEF1         | Rep1     |
| k141_237538  | 471          | HIP_pls       | Rep1     |
| k141_699651  | 149          | gb|GG692639.1 | Rep9     |

**check the reads mapped on to pTEF1, maybe can be mapped on to HIP replicon

1. reads name, sequence mapped on to HIP replicon?? why bwa assign then to pTEF1??

blast??

```
HIP='/ms/11/cong/project/HiC/data/ref/HIP11704'
ASM='/ms/11/cong/project/HiC/ASSEMBLY/MEGAHIT/CoAsm_150PE'
blastn -query ${file} -db ${HIP}/HIP11704 -perc_identity 90 -num_threads 16 -outfmt '6 qseqid sseqid pident mismatch gapopen qlen slen length qstart qend sstart send evalue' -out ${ASM}/blast_result/CoAsm_HIP_genome.tsv

PLASMID_FINDER='/home/cxz163430/software/plasmidfinder'
blastn -query ${file} -db ${PLASMID_FINDER}/plasmidfinder_db/AllRep -perc_identity 90 -num_threads 16 -outfmt '6 qseqid sseqid pident mismatch gapopen qlen slen length qstart qend sstart send evalue' -out ${ASM}/blast_result/CoAsm_rep.tsv
```


* Replicon: No hit
* HIP genome: K00364:132:H2VWMBBXY:4:1120:2696:40631    gb|GG692631.1|, rep9 region?


* very tedious thing!!!

match seq on reads K00364:132:H2VWMBBXY:4:2202:32086:2615@k141_2242115: GATCGATCATTAAAGTAAAAAAAGAAGAAAAAGAAAGCTATATAAAGG

can be also mapped to HIP_PLS contigs (k141_237538)



So we only keep those reads on HIP_pls and HIP_chr replicon region.

####Day 13, V583

####Day 13, VE
2. contig abundance


Find res gene on individual assembly and the prokka annotate those res contigs.

* resfinder on mz
* get res contigs in R
*
seqkit grep -f resfinder/0HIP11704_contigs.list 0HIP11704_150PE.contigs.fa > resfinder/0HIP11704_res.contigs.fa
prokka --quiet --cpus $threads --outdir ${out}/prokka_out/$bin_name --prefix $bin_name ${out}/tmp_bin.fa
* PROKKA